staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/admission.go

```diff
@@ -275,11 +275,28 @@ func (a *GenericAdmissionWebhook) callHook(ctx context.Context, h *v1alpha1.Exte
 		return fmt.Errorf("admission webhook %q denied the request without explanation", h.Name)
 	}
 
+	// Add more information about the webhook controller in the response
+	addHookInfo(h.Name, response.Status.Result)
+
 	return &apierrors.StatusError{
 		ErrStatus: *response.Status.Result,
 	}
 }
 
+func addHookInfo(name string, result *metav1.Status) {
+	deniedBy := fmt.Sprintf("admission webhook %q denied the request", name)
+	// Copy status reason to message if message is empty
+	if len(result.Message) == 0 && len(result.Reason) > 0 {
+		result.Message = string(result.Reason)
+	}
+	retry := ""
+	if result.Details != nil && result.Details.RetryAfterSeconds > 0 {
```

Comment:
I don't think "retry seconds" need to be in the message. If client wants to retry, it should interpret the `result.Details.RetryAfterSeconds`.